octobot_trading/personal_data/orders/order_util.py

```diff
@@ -435,3 +437,55 @@ async def get_order_size_portfolio_percent(exchange_manager, order_amount, side,
 
 def generate_order_id():
     return str(uuid.uuid4())
+
+
+async def update_from_order_storage(order, exchange_manager, pending_groups):
+    # only real orders can be updated by stored orders
+    if exchange_manager.is_trader_simulated or exchange_manager.is_backtesting \
+            or not exchange_manager.storage_manager.orders_storage:
+        return
+    order_details = await exchange_manager.storage_manager.orders_storage.get_startup_order_details(order.order_id)
+    if order_details:
+        await _update_from_order_details(order, order_details, exchange_manager, pending_groups)
+
+
+async def create_order_from_order_storage_details(order_storage_details, exchange_manager, pending_groups):
+    order = orders_storage.create_order_from_storage_details(
+        exchange_manager.trader,
+        order_storage_details
+    )
+    await _update_from_order_details(order, order_storage_details, exchange_manager, pending_groups)
+    return order
+
+
+async def _update_from_order_details(order, order_details, exchange_manager, pending_groups):
+    # rebind order attributes that are not stored on exchange
+    order_dict = order_details[orders_storage.OrdersStorage.ORIGIN_VALUE_KEY]
+    order.tag = order_dict[enums.ExchangeConstantsOrderColumns.TAG.value]
+    order.trader_creation_kwargs = order_details[enums.PersistedOrdersAttr.TRADER_CREATION_KWARGS.value]
+    order.exchange_creation_params = order_details[enums.PersistedOrdersAttr.EXCHANGE_CREATION_PARAMS.value]
+    order.set_shared_signal_order_id(order_details[enums.PersistedOrdersAttr.SHARED_SIGNAL_ORDER_ID.value])
+    order.has_been_bundled = order_details[enums.PersistedOrdersAttr.HAS_BEEN_BUNDLED.value]
+    order.associated_entry_ids = order_details[enums.PersistedOrdersAttr.ENTRIES.value]
```

Comment:
Shouldn't we add default values if the key doesn't exist?

Reply:
All those keys are always filled in format_order. To me adding default values is not required

Reply:
So we accept that this code can raise a key error which prevents the order from been loaded ? It could be for any reason like a format change

Reply:
you are right, i'll make it so that it doest crash if the format changes